retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
> Never attaching an `instance` label so we can forget about the whole thing after ingesting a metric from PGW.

For the PGW we do want `instance` labels on some metrics, namely the metrics of the PGW itself.

The cases I can think of boil down to 1 and 2, a boolean flag is fine.
I'd rather not have an every growing number of modes for various exporters, rather design something that's generic enough for anyone to handle.